pkg/pillar/types/global_test.go

```diff
@@ -0,0 +1,290 @@
+		t.Logf("Running test case %s", testname)
+		(&specMap).AddIntItem(test.key, test.defaultInt, test.min, test.max)
+		assert.Equal(t, test.expectedVal, specMap.GlobalSettings[test.key].IntDefault)
+	}
+}
+func TestAddBoolItem(t *testing.T) {
+	specMap := ConfigItemSpecMap{}
+	specMap.GlobalSettings = make(map[GlobalSettingKey]ConfigItemSpec)
+	testMatrix := map[string]struct {
+		key         GlobalSettingKey
+		defaultBool bool
+		expectedVal bool
+	}{
+		"Test True": {
+			key:         MintimeUpdateSuccess,
+			defaultBool: true,
+			expectedVal: true,
+		},
+		"Test False": {
+			key:         MetricInterval,
+			defaultBool: false,
+			expectedVal: false,
+		},
+	}
+	for testname, test := range testMatrix {
+		t.Logf("Running test case %s", testname)
+		(&specMap).AddBoolItem(test.key, test.defaultBool)
+		assert.Equal(t, test.expectedVal, specMap.GlobalSettings[test.key].BoolDefault)
+	}
+}
+func TestAddTriStateItem(t *testing.T) {
+	specMap := ConfigItemSpecMap{}
+	specMap.GlobalSettings = make(map[GlobalSettingKey]ConfigItemSpec)
+	testMatrix := map[string]struct {
+		key             GlobalSettingKey
+		defaultTristate TriState
+		expectedVal     TriState
+	}{
+		"Test None": {
+			key:             ConfigInterval,
+			defaultTristate: TS_NONE,
+			expectedVal:     TS_NONE,
+		},
+		"Test Enabled": {
+			key:             MetricInterval,
+			defaultTristate: TS_ENABLED,
+			expectedVal:     TS_ENABLED,
+		},
+		"Test Disabled": {
+			key:             MintimeUpdateSuccess,
+			defaultTristate: TS_DISABLED,
+			expectedVal:     TS_DISABLED,
+		},
+	}
+	for testname, test := range testMatrix {
+		t.Logf("Running test case %s", testname)
+		(&specMap).AddTriStateItem(test.key, test.defaultTristate)
+		assert.Equal(t, test.expectedVal, specMap.GlobalSettings[test.key].TriStateDefault)
+	}
+}
+func TestAddStringItem(t *testing.T) {
+	specMap := ConfigItemSpecMap{}
+	specMap.GlobalSettings = make(map[GlobalSettingKey]ConfigItemSpec)
+	testMatrix := map[string]struct {
+		key           GlobalSettingKey
+		defaultString string
+		expectedVal   string
+	}{
+		"Test None": {
+			key:           ConfigInterval,
+			defaultString: "info",
+			expectedVal:   "info",
+		},
+	}
+	for testname, test := range testMatrix {
+		t.Logf("Running test case %s", testname)
+		(&specMap).AddStringItem(test.key, test.defaultString, parseLevel)
+		assert.Equal(t, test.expectedVal, specMap.GlobalSettings[test.key].StringDefault)
+	}
+}
+func TestParseGlobalItem(t *testing.T) {
+	specMap := ConfigItemSpecMap{}
+	specMap.GlobalSettings = make(map[GlobalSettingKey]ConfigItemSpec)
+	specMap.GlobalSettings[ConfigInterval] = ConfigItemSpec{
+		ItemType:        ConfigItemTypeString,
+		StringValidator: parseLevel,
+	}
+	specMap.GlobalSettings[MetricInterval] = ConfigItemSpec{
+		ItemType: ConfigItemTypeInt,
+	}
+	specMap.GlobalSettings[MintimeUpdateSuccess] = ConfigItemSpec{
+		ItemType: ConfigItemTypeBool,
+	}
+	specMap.GlobalSettings[DefaultLogLevel] = ConfigItemSpec{
```

Comment:
loglevel is a string and should use the parseLevel validator.
remoteLogLevel is missing.

Oh, this is test code, but still utterly confusing.